File: PokeApp/PokeApp/ViewController.m
//
//  ViewController.m
//  PokeApp
//
//  Created by Antonio Alves on 9/21/15.
//  Copyright © 2015 Antonio Alves. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController




- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.pokeTable.delegate = self;
    [self req];

    self.ps = [[NSMutableArray alloc] init];
    
    
    
    //Image to background
    [self.view setFrame:CGRectMake(0, 0, self.view.frame.size.width, self.view.frame.size.height)];
    self.view.backgroundColor = [UIColor colorWithPatternImage:[self imageResize:[UIImage imageNamed:@"bg1.png"] andResizeTo:self.view.frame.size]];
    self.pokeTable.backgroundColor = [UIColor clearColor];
    
    
    //Image on Navigation Controller
    UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0.0f, 0.0f, 100.0f, 40.0f)];
    imageView.contentMode = UIViewContentModeScaleAspectFit;
    UIImage *image = [UIImage imageNamed:@"logo.png"];
    [imageView setImage:image];
    self.navigationItem.titleView = imageView;

    

    
    
}
-(UIImage *)imageResize :(UIImage*)img andResizeTo:(CGSize)newSize
{
    CGFloat scale = [[UIScreen mainScreen]scale];
    
    //UIGraphicsBeginImageContext(newSize);
    UIGraphicsBeginImageContextWithOptions(newSize, NO, scale);
    [img drawInRect:CGRectMake(0,0,newSize.width,newSize.height)];
    UIImage* newImage = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return newImage;
}


- (void)req {
        
    NSMutableArray *array = [NSMutableArray array];
    for(int i=1; i<151; i++) {
        [array addObject:@(i)];
    }
    NSMutableArray *ar = [[NSMutableArray alloc] init];
    for (NSNumber *n in array) {
        ar = [self loadPokemonsAtIndex:n];
    }
}


- (NSMutableArray *)loadPokemonsAtIndex:(NSNumber *)index {
    
    NSString *urlString = [NSString stringWithFormat:@"http://pokeapi.co/api/v1/pokemon/%@/", index];
    NSURL *url = [NSURL URLWithString:urlString];
    NSURLRequest *request = [NSURLRequest requestWithURL:url];
    
    
    
    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
    operation.responseSerializer = [AFJSONResponseSerializer serializer];
    
    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation * _Nonnull operation, id  _Nonnull responseObject) {
        NSDictionary *p = responseObject;
        Pokemon *pokemon = [[Pokemon alloc] init];
        pokemon.name = [p objectForKey:@"name"];
        pokemon.specie = [p objectForKey:@"species"];
        pokemon.aid = [[p objectForKey:@"national_id"] intValue];
        
        NSArray *types = [p objectForKey:@"types"];
        NSDictionary *typeName = [types objectAtIndex:0];
        pokemon.type = [typeName objectForKey:@"name"];
        [self.ps addObject:pokemon];
        [self.pokeTable reloadData];
        
    } failure:^(AFHTTPRequestOperation * _Nonnull operation, NSError * _Nonnull error) {
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Falha" message:@"Error na requisição" delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil];
        [alert show];
    }];
    
    [operation start];
    return self.ps;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {

    return [self.ps count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    
    NSInteger row = indexPath.row;
    
    Pokemon *poke = [self.ps objectAtIndex:row];
    
    cell.textLabel.text = poke.name;
    
    cell.imageView.image = [UIImage imageNamed:@"pokeball2"];
    cell.backgroundColor = [UIColor clearColor];
    cell.textLabel.textColor = [UIColor whiteColor];
    cell.detailTextLabel.textColor = [UIColor whiteColor];
    return cell;
}


-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier isEqualToString:@"detail"]) {
        NSIndexPath *indexPath = [self.pokeTable indexPathForSelectedRow];
        DetailViewController *detailViewController = (DetailViewController *)segue.destinationViewController;
        Pokemon *pokeDetail = [self.ps objectAtIndex:indexPath.row];
        detailViewController.detailaid = [NSString stringWithFormat:@"#00%u", pokeDetail.aid];
        detailViewController.detailN = pokeDetail.name;
        detailViewController.detailSpecies = pokeDetail.specie;
        detailViewController.detailTypes = pokeDetail.type;
        detailViewController.title = pokeDetail.name;
    }
}




@end

File: pokeapp/Controllers/ViewController.swift
//
//  ViewController.swift
//  pokeapp
//
//  Created by Maik on 26/04/17.
//  Copyright © 2017 Maik. All rights reserved.
//

import UIKit
import Alamofire
import SwiftyJSON
import ObjectMapper

    class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
        
        @IBOutlet var tblJSON: UITableView!
        var nomes = [String]()
        var urls = [String]()
        var urlPokemon:String?
        
        override func viewDidLoad() {
            super.viewDidLoad()
            
            let logo = UIImage(named: "logo.png")
            let imageView = UIImageView(image:logo)
            self.navigationItem.titleView = imageView
            
            self.view.backgroundColor = UIColor(patternImage: UIImage(named: "bg1.png")!)
            
            Alamofire.request("https://pokeapi.co/api/v1/pokedex/1/").responseJSON { response in
                
                if let JSON = response.result.value {
                    
                    if let dict:Dictionary<String, Any> = JSON as? Dictionary<String, Any>{
                        
                        if let itens = Mapper<Pokemon>().map(JSONObject: dict){
                            
                            for poke in itens.pokemon!{
                                self.nomes.append(poke.name!)
                                self.urls.append(poke.resource_uri!)
                            }
                            
                        }
                        
                    }
                    
                }
                
                self.tblJSON.reloadData()
                
            }
            
        }
        
        override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
            
            let view = segue.destination as! PokemonController
            
            if let indexPath = self.tblJSON.indexPathForSelectedRow {
                let url = urls[indexPath.row]
                view.url = url
            }
            
        }
        
        func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
            return nomes.count
        }
        
        func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
            let nome = nomes[indexPath.row]
            let cell = UITableViewCell(style: UITableViewCellStyle.default, reuseIdentifier: nil)
            cell.textLabel!.text = nome
            return cell
        }
        
        func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
            
            if tableView.cellForRow(at: indexPath) != nil {
                
                self.performSegue(withIdentifier: "pokemonDetalhes", sender: self.urls[indexPath.row])
                
            }
        }
        
        func tableView(_ willDisplayforRowAttableView: UITableView, willDisplay cell: UITableViewCell,
                       forRowAt indexPath: IndexPath) {
            cell.textLabel?.textColor = UIColor.white
            cell.contentView.backgroundColor = UIColor.clear
            cell.backgroundColor = UIColor.clear
            tblJSON.backgroundColor = UIColor.clear
        }
        
    }
